Server/ConcordiaCurriculumManager/Services/CourseService.cs

```diff
@@ -369,14 +368,30 @@ public async Task<ICollection<CourseVersion>> GetCourseVersions(Dossier dossier)
         return currentCourseVersions;
     }
 
-    private async Task VerifyCourseIsNotInCourseGroupingOrThrow(Course course)
+    private async Task VerifyCourseIsNotInCourseGroupingOrThrow(Dossier dossier, Course course)
     {
         var grouping = await _courseGroupingRepository.GetCourseGroupingContainingCourse(course);

```

Comment:
This line was originally written not considering the groupings in the dossier, so if it found any grouping it would throw. Since now we're considering groupings in the dossier, we should use the already existing `GetCourseGroupingsContainingCourse` method to get all the groupings and verify from there (in addition, you also need a check that the grouping you've queried isn't deleted, because a deleted grouping still has references to its courses)